src/Codec/Xlsx/Formatted.hs

```diff
@@ -236,6 +238,25 @@ formatted cs styleSheet =
         , formattedMerges     = reverse (finalSt ^. formattingMerges)
         }
 
+formatSingleWorksheet :: Map (Int, Int) FormattedCell -> StyleSheet -> (Worksheet, StyleSheet)
+formatSingleWorksheet fcs styleSheet = (ws, styleSheet')
+  where
+    initSt        = stateFromStyleSheet styleSheet
+    (cs, finalSt) = runState (mapM (uncurry formatCell) (M.toList fcs)) initSt
+    styleSheet'   = updateStyleSheetFromState styleSheet finalSt
+    ws            = def & wsCells  .~ (M.fromList (concat cs))
+                        & wsMerges .~ (reverse (finalSt ^. formattingMerges))
+                        
+formatWorksheets :: [Map (Int, Int) FormattedCell] -> StyleSheet -> ([Worksheet], StyleSheet)
+formatWorksheets [] styleSheet         = ([], styleSheet)
+formatWorksheets [fcs] styleSheet      = ([ws], styleSheet')
+  where
+    (ws, styleSheet') = formatSingleWorksheet fcs styleSheet
+formatWorksheets (fcs:fcss) styleSheet = (ws:wss, styleSheet'')
+  where
+    (ws,  styleSheet')  = formatSingleWorksheet fcs styleSheet
```

Comment:
please don't do manual recursion when the same could be trivially done with a fold